html/js/product-multilingual.js

```diff
@@ -427,16 +427,26 @@ function update_nutrition_image_copy() {
 
 function update_display(imagefield, first_display) {
 
-    var display_url = imagefield_url[imagefield];
+    var flag=0;
```

Comment:
It's a bit strange to use the imagefield to specify that the image is protected, and then removing "protect_" etc.

Could you just add a parameter to update_display() ?   update_display(imagefield, protected, first_display). It would be clearer that way I think, and we would not need to have the flag, modifiedimagefield etc.